infrastructure/derive/src/lib.rs

```diff
@@ -33,3 +35,108 @@ pub fn derive_hashable_ordering(tokens: TokenStream) -> TokenStream {
     };
     gen.into()
 }
+
+/// This macro will provide a hasable implementation to the a given struct
+#[proc_macro_derive(Hashable, attributes(Digest, hash))]
+pub fn derive_hashable(tokens: TokenStream) -> TokenStream {
+    let input = parse_macro_input!(tokens as DeriveInput);
+    let object_name = &input.ident;
+    let mut digest = None;
+    for attr in &input.attrs {
+        match attr.interpret_meta().unwrap() {
+            syn::Meta::NameValue(val) => {
+                if val.ident.to_string() == "Digest" {
+                    if let syn::Lit::Str(lit) = &val.lit {
+                        digest = Some(lit.value());
+                    }
+                }
+            },
+            _ => (),
+        };
+    }
+    let item = input.data;
+    let fields_text = handle_fields_for_hashable(&item);
+
+    let digest = digest.expect("Could not find Digest attribute"); // this is for the error, if the Digest was not given, this error message will be displayed
+    let varname = Ident::new(&digest, Span::call_site());
+    let gen = quote! {
+        impl  Hashable for #object_name  {
+            fn hash(&self) -> Vec<u8> {
+                let mut hasher = <#varname>::new();
+                #fields_text
+                hasher.result().to_vec()
+            }
+        }
+    };
+    gen.into()
+}
+
+fn handle_fields_for_hashable(item: &Data) -> proc_macro2::TokenStream {
+    match item {
+        Data::Struct(ref item) => {
+            match item.fields {
+                Fields::Named(ref fields) => {
+                    let recurse = fields.named.iter().map(|f| {
+                        let mut do_we_ignore_field = false;
+                        for attr in &f.attrs {
+                            match attr.interpret_meta().unwrap() {
+                                syn::Meta::NameValue(ref val) => {
+                                    if val.ident.to_string() == "hash" {
+                                        if let syn::Lit::Str(lit) = &val.lit {
+                                            if lit.value() == "Hashable_ignore" {
```

Comment:
If I understand this correctly - the usage is

```rust
#[derive(Hashable)]
struct MyThing { 
  #[hash(Hashable_ignore)]
  foo: String,
}
```

Perhaps a better syntax would be

```rust
#[derive(Hashable)]
struct MyThing {
  #[hash(ignore)] /* or */ #[hashable(ignore)]
  foo: String,
}
```

Reply:
agreed, went with :     `#[Hashable(Ignore)]`